plenum/server/monitor.py

```diff
@@ -26,47 +26,77 @@
         self.throughput = 0
-        self.inner_window = inner_window
-        self.min_cnt = min_cnt
+        self.throughput_window_size = throughput_window_size
+        self.throughput_min_cnt = throughput_min_cnt
         self.first_ts = time.perf_counter()
         self.window_start_ts = self.first_ts
-        self.alpha = 2 / (self.min_cnt + 1)
+        self.alpha = 2 / (self.throughput_min_cnt + 1)
 
     def add_request(self, ordered_ts):
         self.update_time(ordered_ts)
         self.reqs_in_window += 1
 
-    def _moving_average(self, next_val):
+    def _accumulate(self, old_accum, next_val):
         """
         Implement exponential moving average
         """
-        return self.throughput * (1 - self.alpha) + next_val * self.alpha
+        return old_accum * (1 - self.alpha) + next_val * self.alpha
 
     def update_time(self, current_ts):
-        while current_ts >= self.window_start_ts + self.inner_window:
-            self.throughput = self._moving_average(self.reqs_in_window / self.inner_window)
-            self.window_start_ts = self.window_start_ts + self.inner_window
+        while current_ts >= self.window_start_ts + self.throughput_window_size:
+            self.throughput = self._accumulate(self.throughput, self.reqs_in_window / self.throughput_window_size)
+            self.window_start_ts = self.window_start_ts + self.throughput_window_size
             self.reqs_in_window = 0
 
-    def get_avg_latency(self):
-        pass
-
-    def get_max_latency(self):
-        pass
-
     def get_throughput(self, request_time):
-        if request_time < self.first_ts + (self.inner_window * self.min_cnt):
+        if request_time < self.first_ts + (self.throughput_window_size * self.throughput_min_cnt):
             return None
         self.update_time(request_time)
-        return self._moving_average(self.reqs_in_window / self.inner_window)
+        return self.throughput
+
+
+class LatencyMeasurement:
+    """
+    Measure latency params
+    """
+
+    def __init__(self, min_latency_count=50, alpha=0.2):
+        self.min_latency_count = min_latency_count
+        # map of client identifier and (total_reqs, avg_latency) tuple
+        self.avg_latencies = {}    # type: Dict(str, (int, float))
+        self.alpha = alpha
+
+    def add_duration(self, identifier, duration):
+        if identifier not in self.avg_latencies:
+            self.avg_latencies[identifier] = (0, .0)
+        total_reqs, curr_avg_lat = self.avg_latencies[identifier]
+        total_reqs += 1
+        self.avg_latencies[identifier] = (total_reqs,
+                                          self._accumulate(curr_avg_lat,
+                                                           duration))
+
+    def _accumulate(self, old_accum, next_val):
+        """
+        Implement exponential moving average
+        """
+        return old_accum * (1 - self.alpha) + next_val * self.alpha
+
+    def get_avg_latency(self, identifier):
```

Comment:
Should we include `get_avg_latency()` (for all clients) into this class to handle `min_count` for all clients, not individually?